diff --git a/Makefile b/Makefile
index 17bcda6..519394d 100644
--- a/Makefile
+++ b/Makefile
@@ -85,7 +85,7 @@ CMAKE_ANDROID_FLAGS := \
 	-DCMAKE_POSITION_INDEPENDENT_CODE=ON \
 	-DCMAKE_BUILD_TYPE=Release
 
-MODULES := "augain;aaudio;dtls_srtp;opus;g711;libg722;g7221;codec2;amr;gzrtp;stun;turn;ice;presence;mwi;account;natpmp;srtp;uuid;sndfile;mixminus;debug_cmd"
+MODULES := "augain;aaudio;opensles;dtls_srtp;opus;g711;libg722;g7221;codec2;amr;gzrtp;stun;turn;ice;presence;mwi;account;natpmp;srtp;uuid;sndfile;mixminus;debug_cmd"
 
 APP_MODULES := "g729;ilbc"
 
@@ -231,7 +231,10 @@ libre.a: Makefile
 		$(CMAKE_ANDROID_FLAGS) \
 		-DCMAKE_FIND_ROOT_PATH="$(NDK_PATH);$(PWD)/openssl" \
 		-DOPENSSL_VERSION_MAJOR=3 \
-		-DOPENSSL_ROOT_DIR=$(PWD)/openssl && \
+		-DOPENSSL_ROOT_DIR=$(PWD)/openssl \
+		-DOPENSSL_INCLUDE_DIR=$(PWD)/openssl/include \
+		-DOPENSSL_CRYPTO_LIBRARY=$(PWD)/openssl/libcrypto.a \
+		-DOPENSSL_SSL_LIBRARY=$(PWD)/openssl/libssl.a && \
 	cmake --build . --target re -j$(CPU_COUNT)
 
 libbaresip: Makefile amr codec2 g729 g722 g7221 gzrtp ilbc openssl opus sndfile libre.a
@@ -243,10 +246,15 @@ libbaresip: Makefile amr codec2 g729 g722 g7221 gzrtp ilbc openssl opus sndfile
 		-DSTATIC=ON -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON \
 		-DAAUDIO_INCLUDE_DIR=${TOOLCHAIN}/sysroot/usr/include \
 		-DAAUDIO_LIBRARY=${TOOLCHAIN}/sysroot/usr/lib/$(TARGET)/$(API_LEVEL)/libaaudio.so \
+		-DOPENSLES_INCLUDE_DIR=${TOOLCHAIN}/sysroot/usr/include \
+		-DOPENSLES_LIBRARY=${TOOLCHAIN}/sysroot/usr/lib/$(TARGET)/$(API_LEVEL)/libOpenSLES.so \
 		-Dre_DIR=$(PWD)/re/cmake \
 		-DRE_LIBRARY=$(PWD)/re/build/libre.a \
 		-DRE_INCLUDE_DIR=$(PWD)/re/include \
 		-DOPENSSL_ROOT_DIR=$(PWD)/openssl \
+		-DOPENSSL_INCLUDE_DIR=$(PWD)/openssl/include \
+		-DOPENSSL_CRYPTO_LIBRARY=$(PWD)/openssl/libcrypto.a \
+		-DOPENSSL_SSL_LIBRARY=$(PWD)/openssl/libssl.a \
 		-DG729_INCLUDE_DIR=$(PWD)/bcg729/include \
 		-DILBC_INCLUDE_DIR=$(PWD)/ilbc/iLBC_rfc3951 \
 		-DOPUS_INCLUDE_DIR=$(PWD)/opus/include_opus \
